mlflow/tracking/client.py

```diff
@@ -30,14 +32,31 @@ def __init__(self, tracking_uri=None, registry_uri=None):
                              defaults to the service set by ``mlflow.tracking.set_tracking_uri``.
         """
         final_tracking_uri = tracking_uri or utils.get_tracking_uri()
-        self.tracking_client = TrackingServiceClient(final_tracking_uri)
-        try:
-            self.registry_client = ModelRegistryClient(registry_uri or final_tracking_uri)
-        except MlflowException as e:
-            if "Unexpected URI" in e.message:
-                self.registry_client = None
-            else:
-                raise e
+        self._tracking_client = TrackingServiceClient(final_tracking_uri)
+        self._registry_uri = registry_uri or final_tracking_uri
```

Comment:
```
        self._registry_uri = registry_uri or final_tracking_uri
        self._tracking_client = TrackingServiceClient(final_tracking_uri)
        # registry client is not defined. APIs are expected to use _get_registry_client() for resolution.
```

Reply:
Great suggestion! Added a comment to the constructor and re-ordered the variable definitions!